cmd/validate/main.go

```diff
@@ -0,0 +1,228 @@
+		}
+
+		switch kw, params := bits[0], bits[1:]; kw {
+		case logStateVkey:
+			if expect != logStateVkey {
+				return fmt.Errorf("line %d: found vkey keyword, but expected %q", num, expect)
+			}
+			if err := expectParts(logStateVkey, len(params), 1); err != nil {
+				return fmt.Errorf("line %d: %v", num, err)
+			}
+			if _, err := f_note.NewVerifier(params[0]); err != nil {
+				return fmt.Errorf("line %d: invalid vkey - %v", num, err)
+			}
+			expect = logStateOrigin
+			continue
+		case logStateOrigin:
+			if expect != logStateOrigin {
+				return fmt.Errorf("line %d: found origin keyword, but expcted %q", num, expect)
+			}
+			if len(params) == 0 {
+				return fmt.Errorf("line: %d: expected origin parameter, but none found", num)
+			}
+			expect = logStateQPD
+			continue
+		case logStateQPD:
+			// Origin is optional
+			if expect == logStateOrigin {
+				expect = logStateQPD
+			}
+			if expect != logStateQPD {
+				return fmt.Errorf("line %d: found qpd keyword, but expected %q", num, expect)
+			}
+			if err := expectParts(logStateQPD, len(params), 1); err != nil {
+				return fmt.Errorf("line %d: %v", num, err)
+			}
+			if v, err := strconv.ParseInt(params[0], 10, 64); err != nil {
+				return fmt.Errorf("line %d: invalid QPD value - %v", num, err)
+			} else if v <= 0 {
+				return fmt.Errorf("line %d: invalid QPD value <= 0", num)
+			}
+			expect = logStateContact
+			continue
+		case logStateContact:
+			if expect != logStateContact {
+				return fmt.Errorf("line %d: found contact keyword, but expected %q", num, expect)
+			}
+			if len(params) == 0 {
+				return fmt.Errorf("line %d: no contact details provided", num)
+			}
+			expect = logStateVkey
+			continue
+		}
+	}
+	if err := s.Err(); err != nil {
+		return err
+	}
+	if expect != logStateVkey {
+		return fmt.Errorf("incomplete log config at end of file")
+	}
+	return nil
+}
+
+// expectParts returns a descriptive error if got != want.
+// Used above to return helpful errors if an incorrect number of tokens are present on a log config line.
+func expectParts(k string, got, want int) error {
+	if got < want {
+		return fmt.Errorf("missing parameter to %s keyword", k)
+	} else if got > want {
+		return fmt.Errorf("too many parameters to %s keyword", k)
+	}
+	return nil
+}
+
+// lineIter returns an iterator over filtered lines from the provided bytes.
+//
+// The filtering removes whitespace from all lines, and will skip the line entirely if it is either
+// empty, or contains only a comment.
+func lineIter(s *bufio.Scanner) iter.Seq2[string, int] {
+	lineNum := 0
+	return func(yield func(string, int) bool) {
+		for s.Scan() {
+			line := s.Text()
+			lineNum++
+
+			line = strings.TrimSpace(line)
+			// Annoyingly, it seems we can't return nil from stripSplit when the whole line is a comment or the scanner
+			// simply gives up, so handle empty lines here instead.
+			if line == "" {
+				continue
+			}
+			if len(line) > 0 && line[0] == '#' {
```

Comment:
```suggestion
			if line[0] == '#' {
```

I.e., the length check is redundant.